src/test/java/com/fittracker/fittracker/controller/ProductControllerIntegrationTest.java

```diff
@@ -88,6 +98,69 @@ void givenValidGetRequestWithNonexistentId_shouldReturnError() throws Exception
             assertThat(response).isEqualTo(expectedResponse);
         }
 
+
+    }
+
+    @Nested
+    class Put {
+
+        @BeforeEach
+        void beforeEach() {
+            productRepository.save(product());
+        }
+
+        @Test
+        void givenValidPutRequest_shouldUpdateAndReturnProductResponse() throws Exception {
+            String id = productRepository.findAll().iterator().next().getId().toString();
+            var expectedResponse = productResponseWithKcal(300);
+            var expectedProduct = new Product(UUID.fromString(id), 1, "bread", 300, 58, 8, 0,
+                UUID.fromString("948cc727-68e5-455c-ab6d-942e585bde0d"), null, true);
+
+            var response = makeRequestWithBodyAndPathVariable(id, PUT, productRequestWithKcal(300), OK,
+                ProductResponse.class);
+
+            assertThat(response).usingRecursiveComparison().ignoringFields("id").isEqualTo(expectedResponse);
+            assertThat(response.id()).isEqualTo(UUID.fromString(id));
+            var dbProducts = productRepository.findAll();
+            assertThat(dbProducts).hasSize(2);
+            assertThat(findLatestInsertedProduct(dbProducts)).usingRecursiveComparison()
```

Comment:
It's better to avoid logic in tests. Let's remove `findLatestInsertedProduct` and do this instead:
```
assertThat(dbProducts).usingRecursiveFieldByFieldElementComparatorIgnoringFields("updatedAt")
    .containsExactlyInAnyOrder(
        new Product(UUID.fromString(id), 1, "bread", 300, 58, 8, 0,
            UUID.fromString("948cc727-68e5-455c-ab6d-942e585bde0d"), null, true),
        new Product(UUID.fromString(id), 0, "bread", 245, 58, 8, 0,
            UUID.fromString("948cc727-68e5-455c-ab6d-942e585bde0d"),
            LocalDateTime.of(2023, 10, 10, 4, 20), false));
```